k4FWCore/python/k4FWCore/utils.py

```diff
@@ -0,0 +1,159 @@
+
+    This function imports a module from a given filename, with the option to
+    specify the module's name and inject global variables into the module before
+    it is returned. If `module_name` is not provided, the filename is used as
+    the module name after replacing '.' with '_'. Global variables can be passed
+    as a dictionary to `global_vars`, which will be injected into the module's
+    namespace.
+
+    Args:
+        filename (str): The path to the file from which to import the module.
+        module_name (Optional[str]): The name to assign to the module. Defaults
+            to None, in which case the filename is used as the module name.
+        global_vars (Optional[Dict[str, Any]]): A dictionary of global variables
+            to inject into the module's namespace. Defaults to None.
+
+    Returns:
+        Any: The imported module with the specified modifications.
+
+    Raises:
+        FileNotFoundError: If the specified file does not exist.
+        ImportError: If there is an error during the import process.
+
+    """
+    filename = os.path.abspath(filename)
+    if not os.path.exists(filename):
+        raise FileNotFoundError(f"No such file: '{filename}'")
+
+    module_name = module_name or os.path.basename(filename).replace(".", "_")
+    loader = SourceFileLoader(module_name, filename)
+    spec = importlib.util.spec_from_loader(loader.name, loader)
+    module = importlib.util.module_from_spec(spec)
+    if global_vars:
+        module.__dict__.update(global_vars)
+    loader.exec_module(module)
+    return module
+
+
+def load_file(opt_file: Union[TextIOWrapper, str, os.PathLike]) -> None:
+    """Loads and executes the content of a given file in the current interpreter session.
+
+    This function takes a file object or a path to a file, reads its content,
+    and then executes it as Python code within the global scope of the current
+    interpreter session. If `opt_file` is a file handle it will not be closed.
+
+    Args:
+        opt_file (Union[TextIOWrapper, str, os.PathLike]): A file object or a
+                                                           path to the file that
+                                                           contains Python code
+                                                           to be executed.
+
+    Raises:
+        FileNotFoundError: If `opt_file` is a path and no file exists at that path.
+        IOError: If there's an error opening or reading the file.
+        SyntaxError: If there's a syntax error in the code being executed.
+        Exception: Any exception raised by the executed code will be propagated.
+
+    """
+    if isinstance(opt_file, (str, os.PathLike)):
+        with open(opt_file, "r") as file:
+            code = compile(file.read(), file.name, "exec")
+    else:
+        code = compile(opt_file.read(), opt_file.name, "exec")
+
+    exec(code, globals())
+
+
+class SequenceLoader:
```

Comment:
We could make this even more general by naming it `AlgorithmList` (or `AlgoListBuilder`) and then make it behave something like a normal list, but with the additional functionality that is currently present.